util/env_posix.cc

```diff
@@ -767,12 +776,12 @@ class PosixEnv : public Env {
                                  const DBOptions& db_options) const override {
     EnvOptions optimized = env_options;
     optimized.use_mmap_writes = false;
-    optimized.use_direct_writes = false;
+    optimized.use_direct_writes = db_options.use_direct_wal_writes;
     optimized.bytes_per_sync = db_options.wal_bytes_per_sync;
     // TODO(icanadi) it's faster if fallocate_with_keep_size is false, but it
     // breaks TransactionLogIteratorStallAtLastRecord unit test. Fix the unit
     // test and make this false
-    optimized.fallocate_with_keep_size = true;
+    optimized.fallocate_with_keep_size = !db_options.use_direct_wal_writes;
```

Comment:
on a WAL file, disabling "keep size" means i don't have to flush the inode on every append to the log file, and results in considerably improved journaling performance.  note the comment above, which is why im pretty sure im breaking a unit test (@dhruba suggested i post the pull request despite the failure to at least get some comments first).